src/components/tooltip/tooltip-content.tsx

```diff
@@ -29,18 +29,13 @@ export class TooltipContent {
     @Prop({ reflect: true })
     maxlength?: number;
 
-    private isLabelsTextLong = false;
-
-    public componentWillLoad() {
+    public render() {
+        let isLabelsTextLong = false;
```

Comment:
I then added this second fixup where I moved the checks into the render method, so that they would apply correctly if the values changed after the component was first created. This was necessary to make the composite example work.

Reply:
It's possible that it would be more performant to have these checks run in a watcher, rather than for every render, but I'm not so sure, because these checks ought to be quite fast, and a watcher would (I assume) need to set up listeners and stuff.

Reply:
Yeah we don't need to think about performance here I think :slightly_smiling_face: 
Things that are typically slow are network requests and manipulating the DOM and we do neither here. A watcher would require us to introduce some kind of state and state is always evil :smile: